src/main/java/com/google/cloud/genomics/dataflow/pipelines/LoadReadsToBigQuery.java

```diff
@@ -0,0 +1,377 @@
+import com.google.genomics.v1.CigarUnit;
+import com.google.genomics.v1.Read;
+import htsjdk.samtools.ValidationStringency;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.net.URISyntaxException;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.PipelineResult;
+import org.apache.beam.sdk.io.gcp.bigquery.BigQueryIO;
+import org.apache.beam.sdk.options.Default;
+import org.apache.beam.sdk.options.Description;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.options.Validation;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.util.gcsfs.GcsPath;
+import org.apache.beam.sdk.values.PCollection;
+
+/**
+ * Pipeline for loading BAM file (hosted on GCS) into BigQuery.
+ *
+ */
+public class LoadReadsToBigQuery {
+  /*
+   * Pipeline options.
+   */
+  public static interface Options extends GCSOptions, ShardOptions {
+
+    @Description("The Google Cloud Storage path to the BAM file to get reads data from.")
+    @Default.String("")
+    String getBamFilePath();
+    void setBamFilePath(String filePath);
+
+    @Description("For sharded BAM file input, the maximum size in megabytes of a shard processed "
+        + "by a single worker. A non-positive value implies no sharding.")
+    @Default.Integer(100)
+    int getMaxShardSizeMB();
+    void setMaxShardSizeMB(int maxShardSizeMB);
+
+    @Description("Whether to include unmapped mate pairs of mapped reads to match expectations "
+        + "of Picard tools.")
+    @Default.Boolean(false)
+    boolean isIncludeUnmapped();
+    void setIncludeUnmapped(boolean newValue);
+
+    @Description("BigQuery table (in [PROJECT]:[DATASET].[TABLE] format) to store the result.")
+    @Default.String("")
+    String getBigQueryTablePath();
+    void setBigQueryTablePath(String tablePath);
+
+    @Description("Whether to wait until the pipeline completes. This is useful for test purposes.")
+    @Default.Boolean(false)
+    boolean getWait();
+    void setWait(boolean wait);
+
+    @Validation.Required
+    @Description("Google Cloud Storage path to write temp files, if applicable.")
+    String getTempPath();
+    void setTempPath(String tempPath);
+
+    public static class Methods {
+      public static void validateOptions(Options options) {
+        // Validate tempPath.
+        try {
+        // Check that we can parse the path.
+        GcsPath valid = GcsPath.fromUri(options.getTempPath());
+        // GcsPath allows for empty bucket, but that doesn't make for a good temp path.
+        Preconditions.checkArgument(!Strings.isNullOrEmpty(valid.getBucket()), "Bucket must be specified");
+        } catch (Exception x) {
+          Preconditions.checkState(false,
```

Comment:
This swallows the exception rather than presenting it to the user.   Presumably you should at least only catch the expected exception, and it would be best to include it in the message (since you may be denying the user important debugging information, like an access control problem rather than a mispelled path)

Reply:
This still isn't a great user experience:
(1) You're combining two different output methods (one via the flags checking mechanism, that may use stderr instead of stdout, for example) so it's hard to say in what order the user will see the messages.  You would be better to put this as part of the error message passed to the flags library.  Something like:

   Preconditions.checkState(false, "the specified temp path is invalid: " + x.getMessage())

(2) A user doesn't care about the Java type name of the exception, so there isn't a need to print it.

(3) In general, errors should go to standard error, not standard output...  This is true of other places in the code too.